libr/fs/fs.c

```diff
@@ -700,13 +702,22 @@ R_API int r_fs_prompt(RFS* fs, const char* root) {
 	}
 
 	for (;;) {
-		printf ("[%s]> ", path);
-		fflush (stdout);
-		fgets (buf, sizeof (buf) - 1, stdin);
-		if (feof (stdin)) {
+		snprintf (prompt, sizeof (prompt), "[%.*s]> ", sizeof (prompt) - 5, 
+		  path);
+		if (!shell->set_prompt ||
```

Comment:
Its ok if we cant histadd or setprompt. And readline can be defaulted to fgets, dont be that strict, because this repl should work even with shell as null

Reply:
I'm not sure I understood what has to be made in this case. Should I use an "if" to check if shell is NULL and then use fgets instead, at the same time I don't call hist_add?